### DISCUSSION TITLE
vue版本SchemaField组件的schema属性是否可以实现deep watch来刷新组件?

### DISCUSSION THREAD

**Discussion by sage-wen** [alibaba/formily#1618] [Q&A 2.x] [answered]
当我更新表单的Schema时，出现了很多奇怪的问题，一些properties并没有得到更新，
目前只能通过v-if的方式实现强制重新渲染SchemaField组件但这种方式是不推荐使用的，是否有其它更好的方式更新schema?

**Comment by MisicDemone** [comment, marked as answer]
更新schema只会对key有变化的field重新创建实例，key没变化的部分是会复用之前的field实例的。对于这种情况建议使用schema中的x-reactions或form实例中的effects去做联动，如果非要修改schema，可以先置为空对象，在nextTick中再去更新schema

**Reply by sage-wen** [reply]
我看源码中是通过properties的索引来做为key的，是否考虑改用属性的name作为key或其它更好的方式？

**Reply by MisicDemone** [reply]
json-schema中properties的key代表的就是字段名，其实就是Field上的name

**Reply by sage-wen** [reply]
![QQ截图20210622104034](https://user-images.githubusercontent.com/40239058/122854424-4c86b480-d346-11eb-86b9-e366e46936ec.png)

**Reply by sage-wen** [reply]
这是递归组件渲染properties的代码，这边使用properties的索引作为key

**Reply by MisicDemone** [reply]
你说这个key啊...😅

**Reply by MisicDemone** [reply]
> 这是递归组件渲染properties的代码，这边使用properties的索引作为key

这个就欢迎pr了...properties的index确实不太合适

**Comment by Amoslb** [comment]
先置为空对象，在nextTick中再去更新schema ,我用了这个方法貌似还是不行，下一次如果属性名相同，发现dom并没有更新。
